etc/systemd/system-generators/zfs-mount-generator.in

```diff
@@ -191,19 +198,19 @@ Documentation=man:zfs-mount-generator(8)
 DefaultDependencies=no
 Wants=${wants}
 After=${after}
-Before=${before}
```

Comment:
If we do this, should we also drop `Wants=`, `After=`, and `Requires=`? 

The key load may apply to multiple encryption roots, right? So it's inappropriate for any of the particular mount `x-systemd.*` options to apply.  So my vote is to just drop all of these.

Caveat: I just woke up.  So read my arguments very critically.

Reply:
FWIW we had some discussions about this way back when: https://github.com/openzfs/zfs/pull/9649#issuecomment-575934819 (and at least the following comment by @rlaager)

Reply:
@didrocks What is the effect of removing this `Before=` and why did you make that change?

Reply:
> @didrocks What is the effect of removing this Before= and why did you make that change?
This was done in this commit (https://github.com/openzfs/zfs/pull/10477/commits/e80ded66437661e1bc9eb0bed2d555785be0046d). To expand a little bit more, the idea is to delay the home user mount once the keyfile is accessible. The keyfile can be made accessible way later than zfs-mount.service (which is what is in the before= here), like via a pam module on login for instance, but can be also be made accessible during the boot process (single user, plymouth asking for passphrase) or so on.

Basically, the idea is to not necessary link those units to zfs-mount.service, which will try to mount it, then, by dependency chain, try loading the key-load unit which will fail at this point, creating spam in logs.